### ISSUE TITLE
`No stacks match` while using `applications` custom config

### ISSUE THREAD

**Issue by bmorrison-console** [awslabs/landing-zone-accelerator-on-aws#300]
**Describe the bug**

Hello

Thanks for such an awesome project!

We're having trouble getting an `applications` configuration working in our test LZA v1.5.1 instance's customizations-config.yaml because we are consistently getting the following error at cdk deploy time (both locally and in codebuild):

```
2023-10-26 07:30:17.800 | error | toolkit | No stacks match the name(s) example_persona-lza-CustomizationsStack-555555555555-ap-southeast-2
```

(where 555555555555 is the correct workload account we are attempting to deploy the `application` block to).

**To Reproduce**

Given the following configuration in customizations-config.yaml (note: this version is based off of the test config found at https://github.com/awslabs/landing-zone-accelerator-on-aws/blob/main/source/packages/%40aws-accelerator/accelerator/test/configs/all-enabled-ou-targets/customizations-config.yaml, but we see the same error for any `applications:` config. We just included this as it's bound to be most 'correct' /readable for you):

```
applications:
  - name: appA
    vpc: Dev
    deploymentTargets:
      accounts:
        - Network
      excludedRegions:
        - us-west-2
    autoscaling:
      name: appA-asg
      maxSize: 4
      minSize: 1
      desiredSize: 2
      launchTemplate: appA-lt
      healthCheckGracePeriod: 300
      healthCheckType: ELB # EC2|ELB
      targetGroups:
        - appA-nlb-tg
      subnets:
        - Dev-App-A
        - Dev-App-B
    launchTemplate:
      name: appA-lt
      blockDeviceMappings:
        - deviceName: /dev/xvda
          ebs:
            deleteOnTermination: true
            encrypted: false
            kmsKeyId: xxx
        - deviceName: /dev/xvdg
          ebs:
            deleteOnTermination: true
            encrypted: true
            volumeSize: 20
      securityGroups:
        - App
      iamInstanceProfile: EC2-Default-SSM-AD-Role
      imageId: ${ACCEL_LOOKUP::ImageId:/aws/service/ami-amazon-linux-latest/amzn2-ami-hvm-x86_64-gp2}
      instanceType: t3.xlarge
      networkInterfaces:
        - deleteOnTermination: true
          description: secondary network interface
          deviceIndex: 1
          groups:
            - App
          networkCardIndex: 1
          subnetId: Dev-App-A
    targetGroups:
      - name: appA-nlb-tg
        port: 80
        protocol: HTTP
        type: instance
        protocolVersion: HTTP1
        connectionTermination: true
        preserveClientIp: true
        proxyProtocolV2: true
      - name: appA-alb-tg-1
        port: 80
        protocol: HTTP
        type: instance
        connectionTermination: true
        preserveClientIp: true
        proxyProtocolV2: true
        healthCheck:
          enabled: true
          port: 80
          protocol: HTTP
    networkLoadBalancer:
      name: appA-nlb
      scheme: internal
      deletionProtection: true
      subnets:
        - Dev-App-A
        - Dev-App-B
      crossZoneLoadBalancing: true
      listeners:
        - name: appA-listener-1
          port: 80
          protocol: TCP
          targetGroup: appA-nlb-tg
```

**Expected behavior**

We expect all stacks created by the synth above to be deployed successful with no cdk deploy errors.

**Please complete the following information about the solution:**

- [x] Version: 1.5.1

- [x] Region: ap-southeast-2

- [x] Was the solution modified from the version published on this repository?  No

- [x] If the answer to the previous question was yes, are the changes available on GitHub?
- [x] Have you checked your [service quotas](https://docs.aws.amazon.com/general/latest/gr/aws_service_limits.html) for the services this solution uses? Yes
- [x] Were there any errors in the CloudWatch Logs? codebuild errors

**Screenshots**
If applicable, add screenshots to help explain your problem (please **DO NOT include sensitive information**).

**Additional context**

The Customizations codebuild action in our pipeline fails with:

```
[Container] 2023/10/26 07:28:44.809818 Running command if [ ! -z "${ACCELERATOR_STAGE}" ]; then yarn run ts-node --transpile-only cdk.ts synth --stage $ACCELERATOR_STAGE --require-approval never --config-dir $CODEBUILD_SRC_DIR_Config --partition aws; fi
yarn run v1.22.19
$ /codebuild/output/src1811/src/s3/00/source/packages/@aws-accelerator/accelerator/node_modules/.bin/ts-node --transpile-only cdk.ts synth --stage customizations --require-approval never --config-dir /codebuild/output/src1811/src/s3/01 --partition aws
Done in 70.74s.

[Container] 2023/10/26 07:29:57.414477 Running command if [ "diff" != "${CDK_OPTIONS}" ]; then yarn run ts-node --transpile-only cdk.ts --require-approval never $CDK_OPTIONS --config-dir $CODEBUILD_SRC_DIR_Config --partition aws --app cdk.out; fi
yarn run v1.22.19
$ /codebuild/output/src1811/src/s3/00/source/packages/@aws-accelerator/accelerator/node_modules/.bin/ts-node --transpile-only cdk.ts --require-approval never deploy --stage customizations --config-dir /codebuild/output/src1811/src/s3/01 --partition aws --app cdk.out

✨  Synthesis time: 0.02s

example_persona-lza-CustomizationsStack-111111111111-ap-southeast-2: deploying... [1/1]
example_persona-lza-CustomizationsStack-111111111111-ap-southeast-2: creating CloudFormation changeset...

 ✅  example_persona-lza-CustomizationsStack-111111111111-ap-southeast-2 (no changes)

✨  Deployment time: 1.08s

Stack ARN:
arn:aws:cloudformation:ap-southeast-2:111111111111:stack/example_persona-lza-CustomizationsStack-111111111111-ap-southeast-2/3fb85d80-6d81-11ee-a1e4-0a808fb4eb49

✨  Total time: 1.1s

✨  Synthesis time: 0.01s

Assuming role arn:aws:iam::222222222222:role/AWSControlTowerExecution for 3600 seconds

✨  Synthesis time: 0.01s

Assuming role arn:aws:iam::333333333333:role/AWSControlTowerExecution for 3600 seconds

✨  Synthesis time: 0.01s

Assuming role arn:aws:iam::444444444444:role/AWSControlTowerExecution for 3600 seconds
example_persona-lza-CustomizationsStack-333333333333-ap-southeast-2: deploying... [1/1]
example_persona-lza-CustomizationsStack-222222222222-ap-southeast-2: deploying... [1/1]
example_persona-lza-CustomizationsStack-444444444444-ap-southeast-2: deploying... [1/1]
example_persona-lza-CustomizationsStack-222222222222-ap-southeast-2: creating CloudFormation changeset...
example_persona-lza-CustomizationsStack-333333333333-ap-southeast-2: creating CloudFormation changeset...
example_persona-lza-CustomizationsStack-444444444444-ap-southeast-2: creating CloudFormation changeset...

 ✅  example_persona-lza-CustomizationsStack-222222222222-ap-southeast-2 (no changes)

✨  Deployment time: 0.66s

Stack ARN:
arn:aws:cloudformation:ap-southeast-2:222222222222:stack/example_persona-lza-CustomizationsStack-222222222222-ap-southeast-2/76c13cc0-7236-11ee-82bc-029dede7538b

✨  Total time: 0.66s

 ✅  example_persona-lza-CustomizationsStack-333333333333-ap-southeast-2 (no changes)

✨  Deployment time: 0.72s

Stack ARN:
arn:aws:cloudformation:ap-southeast-2:333333333333:stack/example_persona-lza-CustomizationsStack-333333333333-ap-southeast-2/4708ec80-6d81-11ee-8116-02e47b4becef

✨  Total time: 0.72s

✨  Synthesis time: 0.01s

Assuming role arn:aws:iam::666666666666:role/AWSControlTowerExecution for 3600 seconds

✨  Synthesis time: 0.01s

Assuming role arn:aws:iam::222222222222:role/AWSControlTowerExecution for 3600 seconds
example_persona-lza-CustomizationsStack-666666666666-ap-southeast-2: deploying... [1/1]
2023-10-26 07:30:17.800 | error | toolkit | No stacks match the name(s) example_persona-lza-CustomizationsStack-555555555555-ap-southeast-2
Deployment failed
error Command failed with exit code 1.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.

[Container] 2023/10/26 07:30:17.826459 Command did not exit successfully if [ "diff" != "${CDK_OPTIONS}" ]; then yarn run ts-node --transpile-only cdk.ts --require-approval never $CDK_OPTIONS --config-dir $CODEBUILD_SRC_DIR_Config --partition aws --app cdk.out; fi exit status 1
[Container] 2023/10/26 07:30:17.829554 Phase complete: BUILD State: FAILED
[Container] 2023/10/26 07:30:17.829565 Phase context status code: COMMAND_EXECUTION_ERROR Message: Error while executing command: if [ "diff" != "${CDK_OPTIONS}" ]; then yarn run ts-node --transpile-only cdk.ts --require-approval never $CDK_OPTIONS --config-dir $CODEBUILD_SRC_DIR_Config --partition aws --app cdk.out; fi. Reason: exit status 1
```

In the above output, the account ID 555555555555 maps to our `Network` account, so the correct `account` in the  `AppConfigItem` is being resolved to the correct account ID. The `synth` stage of the build worked fine, however at deploy time we received the following error from CDK:

```
2023-10-26 07:30:17.800 | error | toolkit | No stacks match the name(s) example_persona-lza-CustomizationsStack-555555555555-ap-southeast-2
```

It is worth noting we have no problems with deploying custom stacks via a `customizations` block. Canned CFN templates are being consumed fine, we only receive this error whenever we specify an `applications` block.

Please note we are brand new to the LZA solution so this may well be an issue with our configuration and/or test LZA setup. However, we're struggling to find a way passed the above error. Possible there something wrong in our configuration to cause this?

Fortunately, I was able to re-produce the error locally. I'll add some simple analysis of what we see going on locally as a comment on this Issue, in case that helps with diagnosing what's going wrong here.

Thanks again!

**Comment by bmorrison-console** [comment]
As mentioned we can reproduce the behaviour we see in codebuild locally. Here's the reproducation locally with some extra logging enabled, if it helps.

We can synth fine:

```
(.venv) brad@brads-laptop ~/src/landing-zone-accelerator-on-aws/source/packages/@aws-accelerator/accelerator git((HEAD detached at v1.5.1))
└─ $ MAX_CONCURRENT_STACKS=1 ACCELERATOR_PREFIX=example_persona-lza  DEBUG=debug ./node_modules/.bin/ts-node --transpile-only  ./cdk.ts synth --stage customizations --config-dir /home/bra
d/src/example_persona-lza-config --partition aws --account 555555555555 --region ap-southeast-2
2023-10-26 18:00:02.235 | info | replacements-config | Loading replacements config substitution values
2023-10-26 18:00:02.253 | info | toolkit | Executing cdk synth customizations for aws://555555555555/ap-southeast-2
2023-10-26 18:00:26.675 | info | app | Begin Accelerator CDK App
2023-10-26 18:00:26.683 | info | replacements-config | Loading replacements config substitution values
2023-10-26 18:00:26.701 | info | replacements-config | Loading replacements config substitution values
2023-10-26 18:00:28.120 | info | replacements-config | Loading replacements config substitution values
2023-10-26 18:00:34.473 | info | stack-utils | Stack in account 555555555555 and region ap-southeast-2 using CliCredentialSynthesizer
2023-10-26 18:00:34.499 | info | example_persona-lza-CustomizationsStack-555555555555-ap-southeast-2 | Completed stack synthesis
2023-10-26 18:00:34.500 | info | custom-stack | [custom-stack] Network account explicitly included
2023-10-26 18:00:34.500 | info | stack-utils | Stack in account 555555555555 and region ap-southeast-2 using CliCredentialSynthesizer
2023-10-26 18:00:34.694 | info | example_persona-lza-App-appA-555555555555-ap-southeast-2 | Completed stack synthesis
2023-10-26 18:00:34.694 | info | app | End Accelerator CDK App
```

And synth output looks as we would hope:

```
(.venv) brad@brads-laptop ~/src/landing-zone-accelerator-on-aws/source/packages/@aws-accelerator/accelerator git((HEAD detached at v1.5.1))
└─ $ ls -l cdk.out/
total 32
-rw-rw-r-- 1 brad brad   20 Oct 26 18:00 cdk.out
drwxrwxr-x 2 brad brad 4096 Oct 26 17:59 example_persona-lza-CustomizationsStack-555555555555-ap-southeast-2
-rw-rw-r-- 1 brad brad 6585 Oct 26 18:00 manifest.json
-rw-rw-r-- 1 brad brad  647 Oct 26 18:00 placeHolder.assets.json
-rw-rw-r-- 1 brad brad  766 Oct 26 18:00 placeHolder.template.json
-rw-rw-r-- 1 brad brad    6 Oct 26 18:00 read.843692.1.lock
-rw-rw-r-- 1 brad brad 1133 Oct 26 18:00 tree.json
(.venv) brad@brads-laptop ~/src/landing-zone-accelerator-on-aws/source/packages/@aws-accelerator/accelerator git((HEAD detached at v1.5.1))
└─ $ ls -l cdk.out/example_persona-lza-CustomizationsStack-555555555555-ap-southeast-2
total 224
-rw-rw-r-- 1 brad brad  3038 Oct 26 18:00 AwsSolutions-example_persona-lza-App-appA-555555555555-ap-southeast-2-NagReport.csv
-rw-rw-r-- 1 brad brad    20 Oct 26 18:00 cdk.out
-rw-rw-r-- 1 brad brad  1813 Oct 26 18:00 example_persona-lza-App-appA-555555555555-ap-southeast-2.assets.json
-rw-rw-r-- 1 brad brad 19619 Oct 26 18:00 example_persona-lza-App-appA-555555555555-ap-southeast-2.template.json
-rw-rw-r-- 1 brad brad   598 Oct 26 18:00 example_persona-lza-CustomizationsStack-555555555555-ap-southeast-2.assets.json
-rw-rw-r-- 1 brad brad   862 Oct 26 18:00 example_persona-lza-CustomizationsStack-555555555555-ap-southeast-2.template.json
-rw-rw-r-- 1 brad brad 99260 Oct 26 18:00 manifest.json
-rw-rw-r-- 1 brad brad   647 Oct 26 18:00 placeHolder.assets.json
-rw-rw-r-- 1 brad brad   766 Oct 26 18:00 placeHolder.template.json
-rw-rw-r-- 1 brad brad     6 Oct 26 16:04 read.608947.1.lock
-rw-rw-r-- 1 brad brad     6 Oct 26 17:59 read.817377.1.lock
-rw-rw-r-- 1 brad brad 69106 Oct 26 18:00 tree.json
```

When we try to deploy, we see the error we see in codebuild:

```
).venv) brad@brads-laptop ~/src/landing-zone-accelerator-on-aws/source/packages/@aws-accelerator/accelerator git((HEAD detached at v1.5.1))
└─ $ MAX_CONCURRENT_STACKS=1 ACCELERATOR_PREFIX=example_persona-lza  DEBUG=debug ./node_modules/.bin/ts-node --transpile-only  ./cdk.ts deploy --stage customizations --config-dir /home/brad/src/example_persona-lza-config --partition aws --account 555555555555 --region ap-southeast-2
2023-10-26 18:04:58.106 | info | replacements-config | Loading replacements config substitution values
2023-10-26 18:04:58.123 | info | toolkit | Executing cdk deploy customizations for aws://555555555555/ap-southeast-2
2023-10-26 18:04:58.200 | info | replacements-config | Loading replacements config substitution values
2023-10-26 18:04:58.204 | info | replacements-config | Loading replacements config substitution values
2023-10-26 18:04:59.730 | info | custom-stack | [custom-stack] Network account explicitly included
2023-10-26 18:04:59.783 | error | toolkit | No stacks match the name(s) example_persona-lza-CustomizationsStack-555555555555-ap-southeast-2
Deployment failed
```


After playing around in the debugger, we've discovered that each of the stacks that are being synth'd are valid, it's just that it appears that the LZA's  call to `cdk deploy` execution is not finding the `example_persona-lza-CustomizationsStack-555555555555-ap-southeast-2` stack. It seems to be there on disk, as we can see from the `ls` output above.


```
$ find cdk.out/ -name "*example_persona-lza-CustomizationsStack-555555555555-ap-southeast-2*"  -ls
  3811031      4 drwxrwxr-x   2 brad     brad         4096 Oct 26 18:04 cdk.out/example_persona-lza-CustomizationsStack-555555555555-ap-southeast-2
  3811038      4 -rw-rw-r--   1 brad     brad          598 Oct 26 18:00 cdk.out/example_persona-lza-CustomizationsStack-555555555555-ap-southeast-2/example_persona-lza-CustomizationsStack-555555555555-ap-southeast-2.assets.json
  3811037      4 -rw-rw-r--   1 brad     brad          862 Oct 26 18:00 cdk.out/example_persona-lza-CustomizationsStack-555555555555-ap-southeast-2/example_persona-lza-CustomizationsStack-555555555555-ap-southeast-2.template.json
```

If there is any more info that'd help, please let us know.

**Comment by bmorrison-console** [comment]
Looks like this was fixed in v1.5.2 (probably by https://github.com/awslabs/landing-zone-accelerator-on-aws/commit/c6c9362538d72b0df176b503f5d2789e678da308). 

 Thanks for the fix! I'll close this issue.

**Comment by bmorrison-console** [comment]
Attempting to close.
